cfd_protocol/src/protocol/transactions.rs

```diff
@@ -496,3 +494,42 @@ pub fn punish_transaction(
 
     Ok(punish_tx)
 }
+
+#[derive(Clone, Debug)]
+struct Fee {
+    fee: f64,
+}
+
+impl Fee {
+    pub fn new(signed_vbytes: f64) -> Self {
+        let fee = signed_vbytes * SATS_PER_VBYTE;
+        Self { fee }
+    }
+
+    /// Note: does not mutate, but creates another Fee object
+    pub fn add(&self, number: f64) -> Fee {
+        Fee {
+            fee: self.fee + number,
+        }
+    }
+
+    pub fn rounded(&self) -> u64 {
+        self.fee.ceil() as u64
+    }
+
+    pub fn split(&self) -> (u64, u64) {
```

Comment:
I am a bit confused by this, can we maybe add some more simple tests to show the behavior?

let's say we have `3.0` as `f64` self:

> `let half = self.rounded() / 2;` 

`self.rounded()` = 3
`half = 3 / 2` = 1.5

> `(half as u64, self.rounded() - half)`

`half as u64` = 2 // I assume? Does this always round?
`self.rounded() - half` = 3 - 2 = 1

Is that the expected behavior?

Would be cool to see behavior for odd and even number, and potentially also the dot ranges.

Reply:
I added a few test cases in addition to the proptest to illustrate the behaviour, for `1.0`, `2.0` and `2.6` - it should be much more clear now :)